### DISCUSSION TITLE
Coal production since 1900: annotated area charts with plotnine

### DISCUSSION THREAD

**Discussion by nrennie** [has2k1/plotnine#836] [2024 Plotnine Contest]
### Authors

Nicola Rennie

### Links

[website](https://nrennie.rbind.io/), [github](https://github.com/nrennie), [twitter](https://x.com/nrennie35), [mastodon](https://fosstodon.org/@nrennie), [bluesky](https://bsky.app/profile/nrennie.bsky.social), [linkedin](https://www.linkedin.com/in/nicola-rennie/)

### Full description

This entry is a tutorial on how to create annotated area charts using `plotnine`. It uses data from Carbon Majors (via a #TidyTuesday dataset) which shows how coal production has changed over time since 1900.

![](https://raw.githubusercontent.com/nrennie/2024-plotnine-contest/main/plot.png)

Fully reproducible, documented tutorial: [nrennie.rbind.io/2024-plotnine-contest/](https://nrennie.rbind.io/2024-plotnine-contest/)

The tutorial is also duplicated as a blog post on my website: [nrennie.rbind.io/blog/plotnine-annotated-area-chart](https://nrennie.rbind.io/blog/plotnine-annotated-area-chart/)

The source code for the tutorial can be found on GitHub: [github.com/nrennie/2024-plotnine-contest](https://github.com/nrennie/2024-plotnine-contest)

### Code repository

GitHub repository: [github.com/nrennie/2024-plotnine-contest](https://github.com/nrennie/2024-plotnine-contest)

**Comment by kyleGrealis** [comment]
what a beautiful plot! great work

**Comment by ahmedmoosman** [comment]
What a nice chart!

**Comment by matt-humphrey** [comment]
This is fantastic, and such a useful reference. Thank you Nicola!

**Comment by has2k1** [comment]
![plot](https://github.com/user-attachments/assets/ded9fcf5-cd60-477a-bca4-6b7384ef3cd8)

**Comment by has2k1** [comment]
@nrennie, The winning submission has been announced [here](https://posit.co/blog/winner-of-the-2024-plotnine-plotting-contest/). Thank you for taking part in the contest.

**Comment by 123-fake-st** [comment]
@nrennie this is a really nice plot.  I appreciate the look and style and the satisfying detail for me is the **Coal types** text which also serves as a legend with the color-coded text.  I'm sure I will look back into your code for inspiration during some future work.
